packages/shared/src/components/tooltips/ChangelogTooltip.tsx

```diff
@@ -0,0 +1,162 @@
+              >
+                New release
+              </Button>
+              <ModalClose
+                onClick={(event) => {
+                  if (typeof onRequestClose === 'function') {
+                    onRequestClose(event);
+                  }
+
+                  dismissChangelog();
+                }}
+                data-testid="changelogModalClose"
+              />
+            </header>
+            <section className="flex flex-col flex-1 p-5 h-full shrink max-h-full">
+              <Image
+                className="object-cover w-44 h-28 rounded-lg"
+                alt="Post Cover image"
+                src={post.image}
+                fallbackSrc={cloudinary.post.imageCoverPlaceholder}
+                loading="lazy"
+                data-testid="changelogImage"
+              />
+              <h3
+                className="mt-2 font-bold text-theme-label-primary typo-title3"
+                data-testid="changelogTitle"
+              >
+                {post.title}
+              </h3>
+              <time
+                className="text-theme-label-quaternary typo-callout"
+                dateTime={post.createdAt}
+                data-testid="changelogDate"
+              >
+                {postDateFormat(post.createdAt)}
+              </time>
+              <div className="mt-2 w text-theme-label-tertiary typo-callout">
+                {post.summary}
+              </div>
+            </section>
+            <footer className="flex gap-3 items-center py-3 px-4 w-full h-16 border-t border-theme-divider-tertiary">
+              <Button
+                className="btn-tertiary"
+                onClick={dismissChangelog}
+                tag="a"
+                href={post.permalink}
+                data-testid="changelogReleaseNotesBtn"
+              >
+                Release notes
+              </Button>
+              {isExtension && (
+                <Button
+                  tag={isFirefoxExtension ? 'a' : undefined}
+                  href={
+                    isFirefoxExtension
+                      ? 'https://r.daily.dev/firefoxupdate'
+                      : undefined
+                  }
+                  className="bg-cabbage-40 btn-primary"
+                  data-testid="changelogExtensionBtn"
+                  onClick={async () => {
+                    await updateExtension();
+                  }}
+                >
+                  Update extension
+                </Button>
+              )}
+            </footer>
+          </div>
+        )
+      }
+      offset={tooltipArrowOffset}
+      interactive
+      container={{
+        className: 'shadow',
+        paddingClassName: 'p-0',
+        roundedClassName: 'rounded-16',
+        bgClassName: 'bg-cabbage-40',
+        arrowClassName:
+          'bg-cabbage-40 bottom-24 !left-0 !h-2.5 !w-0 flex items-center justify-end before:!w-10 before:!h-px before:!transform-none',
```

Comment:
Just a question, are all the important overrides needed here?

Reply:
Yeah, because of the tipsy tooltip.

Reply:
In this case, it would probably be better to turn the `arrow` property to a `ReactNode` so we can pass this custom component instead.But will leave it up to you to decide.

Reply:
I am not a fan of props doing two things or having union types like this because it gets messy real fast. But refactoring `arrow` to ReactNode exclusively  would make sense. I would also prefer to do it outside of this task because it is not required for any core functionality. 

If both of you think we should do it here I am up for it also. 👐

Reply:
Happy for another task :)

Reply:
I agree with that. I think we only had one place where we introduced something similar, I just thought of backward compatibility. Making it `ReactNode` exclusively as you mentioned would be the best option. You will just have to refactor existing implementations with `arrow={false}` to `arrow={<></>}`. So the ones using default would still yield the expected result for deciding whether to display the prop value or the original one.

Reply:
@rebelchris would love to hear what you think.

Reply:
Think it's a minor thing for another task really.
Don't want to overcomplicate this PR in particular.

Reply:
Just my two cents, I believe overrides are okay for one or two classes, but when it becomes many, I think it will defeat the purpose of the original component/element, but the problem is, where do we draw the line where it is still considered acceptable.

Reply:
> Think it's a minor thing for another task really. Don't want to overcomplicate this PR in particular.

True. We can probably do that in a separate PR/task.

Reply:
Added to backlog.